tools/rmb_tester/test_live_nodes.sh

```diff
@@ -36,21 +45,32 @@ trap cleanup SIGHUP	SIGINT SIGQUIT SIGABRT SIGTERM
 set +e
 echo "redis-server starting .."
 
-redis-server --port 6379 2>&1 > /dev/null&
+redis-server --port 6379 > /dev/null 2>&1 &
 sleep 3
+# clear all databases
+echo "Removes all keys in Redis"
+redis-cli -p 6379 FLUSHALL
 set -e
 
+# ensure that RMB is not already running
+if pgrep -x $(basename "$RMB_BIN") > /dev/null; then
+    echo "Another instance of rmb-peer is already running. Killing..."
+    pkill -x $(basename "$RMB_BIN")
+fi
+
+# ensure the MNEMONIC has no leading or trailing spaces
+MNEMONIC="${MNEMONIC#"${MNEMONIC%%[![:space:]]*}"}"; MNEMONIC="${MNEMONIC%"${MNEMONIC##*[![:space:]]}"}"
```

Comment:
I suggest u fail here if MNEMONIC = "", because the result is not clear in this case

Reply:
Done.
Additionally, there was an issue where the script continued running after the cleanup function executed. Essentially, the cleanup function was called after the rmb-peer start timeout within a subshell, so the exit command in the cleanup function did not terminate the main script, and it continued to run.

The script is now capable of properly handling both errors.